### PR TITLE
Make FCM and InstanceID compatible with tvOS

### PR DESCRIPTION
**Pull request by chliangGoogle**

### CODE REVIEW

File: Firebase/Messaging/FIRMessagingContextManagerService.m [hunk 1]
```diff
@@ -130,6 +130,7 @@ + (BOOL)handleContextManagerLocalTimeMessage:(NSDictionary *)message {
 
 + (void)scheduleLocalNotificationForMessage:(NSDictionary *)message
                                      atDate:(NSDate *)date {
+#if TARGET_OS_IOS
```
Comment (morganchen12) [inline Firebase/Messaging/FIRMessagingContextManagerService.m]: If this method does nothing on tvOS, it should be marked `__TVOS_UNAVAILABLE` in the header file.

File: Example/Firebase.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -103,6 +103,18 @@
 		0672F2F31EBBA7D900818E87 /* GoogleService-Info.plist in Resources */ = {isa = PBXBuildFile; fileRef = 0672F2F11EBBA7D900818E87 /* GoogleService-Info.plist */; };
 		069428831EC3B38C00F7BC69 /* 1mb.dat in Resources */ = {isa = PBXBuildFile; fileRef = 069428801EC3B35A00F7BC69 /* 1mb.dat */; };
 		06C24A061EC39BCB005208CA /* FIRStorageIntegrationTests.m in Sources */ = {isa = PBXBuildFile; fileRef = 06121ECA1EC39A0B0008D70E /* FIRStorageIntegrationTests.m */; };
+		0D814F04A2A1092E7D094615 /* Pods_Core_Example_tvOS.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 1E0B9240C347145E903C14FB /* Pods_Core_Example_tvOS.framework */; };
```
Comment (paulb777) [inline Example/Firebase.xcodeproj/project.pbxproj]: Please run `pod deintegrate` before committing to make the review manageable

File: Firebase/Messaging/FIRMessaging.m [hunk 3]
```diff
@@ -466,15 +466,21 @@ - (void)handleIncomingLinkIfNeededFromMessage:(NSDictionary *)message {
...
            sourceApplication:FIRMessagingAppIdentifier()
                   annotation:@{}];
+#pragma clang diagnostic pop
+#endif
   } else if ([appDelegate respondsToSelector:handleOpenURLSelector]) {
+#if TARGET_OS_IOS
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
     [appDelegate application:application handleOpenURL:url];
-#pragma clang diagnostic pop
```
Comment (paulb777) [inline Firebase/Messaging/FIRMessaging.m]: should this line be deleted?

File: Firebase/Messaging/FIRMessagingContextManagerService.m (hunk 1 above)
Comment (chliangGoogle) [reply Firebase/Messaging/FIRMessagingContextManagerService.m]: Even if the header is marked as unavailable, we will need to mark the implementation with this macro.

This is a private implementation and there are quite some private methods which are not available for tvOS. I'm not sure if we do the same for other community support at this point. This CL is just to ensure the code compiles under tvOS. We might start mark __TVOS_UNAVAILABLE in the public API first.

**Comment by paulb777** [comment]
A few more things I noticed trying to run in the branch:

- `GPB_USE_PROTOBUF_FRAMEWORK_IMPORTS` needs to be added to the Preprocessor Macros setting similar to how it is for the iOS Tests target
- The Test scheme should be marked shared and visible in the scheme editor and then the resulting file added to the repo
- Some extra cruft got into the HEADER_SEARCH_PATHS setting. Make it the same as the iOS test target.

File: Firebase/Messaging/FIRMessaging.m (hunk 3 above)
Comment (chliangGoogle) [reply Firebase/Messaging/FIRMessaging.m]: Done.

**Comment by chliangGoogle** [comment]
There's conflict in the file, should I keep adding IID tests and merge later or just submit this CL first?

**Comment by paulb777** [comment]
The conflict needs to be solved before merging.  It looks feasible now after deintegrating.  At a quick glance, it looks like "Accepting Both" for the one conflict will do it.

Also, the new test targets should be added to the "All tvOS" Aggregate target scheme and then they'll be included in CI.

**Comment by paulb777** [comment]
Do `git merge origin/master` and `git mergetool`

**Comment by chliangGoogle** [comment]
Done merging the conflicts.

**Comment by chliangGoogle** [comment]
It doesn't let me merge by running git merge origin/master
So I did the step 1 of comment line instructions and push the change to this PR.

**Comment by paulb777** [comment]
Looks like we need to update the FirebaseInstanceID version on SpecsStaging for FirebaseMessaging to pass `pod lib lint`.  

To be safe, let's hold off on this until the M44 release branch gets made.

**Pull request merged**